specs/HubProtocol.md

```diff
@@ -48,13 +53,33 @@ On the Callee side, it is up to the Callee's Binder to determine if a method cal
 
 On the Caller side, the user code which performs the invocation indicates how it would like to receive the results and it is up the Caller's Binder to determine how to handle the result. If the Caller expects only a single result, but multiple results are returned, the Caller's Binder should yield an error indicating that multiple results were returned. However, if a Caller expects multiple results, but only a single result is returned, the Caller's Binder should yield that single result and indicate there are no further results.
 
+## Completion and Results
+
+A Invocation is only considered completed when the `Completion` message is recevied. Receiving **any** message using the same `Invocation ID` after a `Completion` message has been received for that invocation is considered a protocol error and the recipient should immediately terminate the connection.
```

Comment:
I am not sure about the requirement of terminating the connection. In a general case you would have to keep all invocation ids to know if the message you received is for an invocation that has already completed. Rather I would say that the client MAY terminate a connection or ignore a message if the invocation id does not correspond to any pending invocations.